Hast.Abstractions/Hast.Vitis.Abstractions/Services/VitisHardwareImplementationComposerBuildProvider.cs

```diff
@@ -208,8 +238,6 @@ public sealed class VitisHardwareImplementationComposerBuildProvider
             try { await CollectReportsAsync(hardwareFrameworkPath, context, implementation, hashId); }
             catch (Exception e) { _logger.LogError(e, "Failed to collect reports."); }
         }
-
-        Cleanup(hardwareFrameworkPath, hashId);
```

Comment:
Why is cleanup not necessary here anymore?

Reply:
The `IHardwareImplementationComposerBuildProvider` interface contains a [separate `CleanupAsync` method](https://github.com/Lombiq/Hastlayer-SDK/blob/dev/Hast.Abstractions/Hast.Synthesis.Abstractions/IHardwareImplementationComposerBuildProvider.cs#L44) and [that one should be implemented instead](https://github.com/Lombiq/Hastlayer-SDK/pull/88/commits/d3faa8be9c139bba8c1a87fc9a61f90b09f88b83). It's called [for every required build provider automatically](https://github.com/Lombiq/Hastlayer-SDK/blob/dev/Hast.Abstractions/Hast.Synthesis.Abstractions/VhdlHardwareImplementationComposer.cs#L57-L61). So we shouldn't call it inside the build provider after the build is completed. The removed code here deleted the temporary files created by v++ after Vitis build has finished. But those files are needed for the Zynq build provider that uses this Vitis build provider to generate them. Now the cleanup is called after all build providers have finished.

Reply:
Got it, thanks.